Python/ceval.c

```diff
@@ -5872,6 +5920,73 @@ do_call_core(PyThreadState *tstate,
     return PyObject_Call(func, callargs, kwdict);
 }
 
+/* Fast alternative for non-keyword calls to C functions. */
+Py_LOCAL_INLINE(PyObject *) _Py_HOT_FUNCTION
+call_cfunction(PyThreadState *tstate,
+              _PyAdaptiveEntry *cache0,
+              _PyCallFunctionCache *cache1,
+              PyObject ***pp_stack,
+              Py_ssize_t oparg,
+              int use_tracing)
+{
+#define MAYBE_TRACE(cfunc) if (use_tracing) {C_TRACE(x, cfunc);} else {x = cfunc;}
+
+    PyObject **pfunc = (*pp_stack) - oparg - 1;
+    PyObject *x = NULL, *w;
+    PyObject **stack = (*pp_stack) - oparg;
+
+    PyObject *func = *pfunc; /* Only for tracing purposes */
+    PyObject *self = PyCFunction_GET_SELF(func);
+    PyCFunction cfunc = PyCFunction_GET_FUNCTION(func);
+
+    switch ((_BuiltinCallKinds)cache0->index) {
+        case PYCFUNCTION_NOARGS:
+        case PYCFUNCTION_O:
+            MAYBE_TRACE(cfunc(self, *stack));
```

Comment:
Remove the `MAYBE_TRACE`. If tracing is on, then the non-specialized instruction will be executed.
An `assert(use_tracing == 0)` should do.

Reply:
The assertion seems to fail on the macOS and Ubuntu tests. Do I have to add a guard somewhere or is that not supposed to happen?